lib/classes/local-file-location.js

```diff
@@ -13,25 +13,30 @@ const mime = require('mime-types');
 class LocalFileLocation {
     /**
      * @param {string} path relative path to file on disk
-     * @param {string} cwd current working directory
+     * @param {string} basePath basePath to the file's relative location on disk
      */
-    constructor(path, cwd) {
+    constructor(path, basePath) {
         assert(typeof path === 'string', '"path" must be of type "string"');
-        assert(typeof cwd === 'string', '"cwd" must be of type "string"');
+        assert(
+            typeof basePath === 'string',
+            '"basePath" must be of type "string"',
+        );
+        assert(!isAbsolute(path), '"path" must be a relative path');
+
         /**
          * @type {string} path relative path to file on disk
          */
         this.relative = path;
 
         /**
-         * @type {string} cwd current working directory
+         * @type {string} basePath to the file's relative location on disk
```

Comment:
Whats the difference between this and `this.relative`? Its possibly me finding the wording in this description to sound like  the description for `this.relative`.

Reply:
maybe bad wording on my part there, the absolute path is made up of the basePath + the relative path. So I meant to say there that that is the basePath to the files location on disk